### PR TITLE
Webpack 4 refactor

### PR DESCRIPTION
**Pull request by ScottRocha**
###  Summary

Implement Webpack 4 hooks API.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackhq/csp-html-webpack-plugin/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've written tests to cover the new code and functionality included in this PR.
* [x] I've read, agree to, and signed the [Contributor License Agreement (CLA)](https://cla-assistant.io/slackhq/csp-html-webpack-plugin).

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -88,53 +88,103 @@ class CspHtmlWebpackPlugin {
...
 
             // eslint-disable-next-line no-param-reassign
             htmlPluginData.html = $.html();
 
             return compileCb(null, htmlPluginData);
           }
+        );
+      });
+    } else {
+      compiler.plugin('compilation', compilation => {
```
Comment (AnujRNair) [inline plugin.js]: Would you be able to reduce the code duplication here? Abstracting the main body of the compilation step out would help keep test coverage of the entire project high, and would be better for maintainability

File: plugin.js [hunk 2]
```diff
@@ -88,53 +88,103 @@ class CspHtmlWebpackPlugin {
...
-        (htmlPluginData, compileCb) => {
-          const $ = cheerio.load(htmlPluginData.html, {
-            decodeEntities: false
-          });
-
-          // if not enabled, remove the empty tag
-          if (!this.isEnabled(htmlPluginData)) {
-            $('meta[http-equiv="Content-Security-Policy"]').remove();
+    if (compiler.hooks) {
+      compiler.hooks.compilation.tap('HtmlWebpackHarddisk', compilation => {
```
Comment (AnujRNair) [inline plugin.js]: `HtmlWebpackHarddisk` should be updated to `CspHtmlWebpackPlugin`

File: plugin.js [hunk 3]
```diff
@@ -88,53 +88,103 @@ class CspHtmlWebpackPlugin {
...
+        compilation.hooks.htmlWebpackPluginAfterHtmlProcessing.tapAsync(
+          'HtmlWebpackHarddisk',
```
Comment (AnujRNair) [inline plugin.js]: This should also be updated to match the plugin name

File: plugin.js (hunk 1 above)
Comment (ScottRocha) [reply plugin.js]: Can do. Give me a few.

**Comment by ScottRocha** [comment]
@AnujRNair I have implemented the changes as requested. Let me know if there is anything else blocking this from being approved.

File: plugin.js (hunk 2 above)
Comment (ScottRocha) [reply plugin.js]: This has been updated in the latest commit.

File: plugin.js (hunk 3 above)
Comment (ScottRocha) [reply plugin.js]: This has also been updated in the latest commit.

**Pull request merged**